data/abilities.js

```diff
@@ -2651,6 +2651,20 @@ let BattleAbilities = {
 	},
 	"pastelveil": {
 		shortDesc: "This Pokemon and its allies cannot be poisoned. On switch-in, cures poisoned allies.",
+		onStart(pokemon) {
+			for (const ally of pokemon.allies()) {
+				if (['psn', 'tox'].includes(ally.status)) {
+					this.add('-activate', pokemon, 'ability: Pastel Veil');
+					ally.cureStatus();
+				}
+			}
+		},
+		onUpdate(pokemon) {
+			if (['psn', 'tox'].includes(pokemon.status)) {
+				this.add('-activate', pokemon, 'ability: Pastel Veil');
+				pokemon.cureStatus();
+			}
+		},
 		onAllySwitchIn(pokemon) {
```

Comment:
Is this necessary? I never read anything that mentioned it in @DaWoblefet's research.

Reply:
Not entirely sure what it's doing in particular, but if I'm reading this right, Pastel Veil should announce itself when blocking poison (except for switching in on Toxic Spikes), indicate its source, and indicate the target being cured or being made immune to poison.

Reply:
It cures the ally's status when the ally switches in. Not the Pastel Veil user, they're already active.

Reply:
That behavior is correct too, it should cure an ally's poison on switchin if the other active Pokemon is Pastel Veil.

Reply:
Isn't `onUpdate` enough to cover the others?

Reply:
Oh, `onUpdate` only affects the user.